src/bundle/DependencyInjection/EzPlatformRichTextExtension.php

```diff
@@ -177,4 +183,72 @@ private function validateCustomTemplatesConfig(
             }
         }
     }
+
+    /**
+     * Validate presence of inline Custom Tags in Toolbars.
+     *
+     * @param array $availableSiteAccesses a list of available SiteAccesses
+     * @param array $customTagsConfig Custom Tags configuration
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
+     */
+    private function validateInlineCustomTagToolbarsConfig(
+        array $availableSiteAccesses,
+        array $customTagsConfig,
+        ContainerBuilder $container
+    ): void
+    {
+        $customTags = $this->getInlineCustomTags($customTagsConfig);
+        foreach ($this->getToolbarsBySiteAccess($availableSiteAccesses, $container) as $siteAccess => $toolbar) {
+            foreach ($toolbar as $toolbarName => $toolbarContent) {
+                // "text" toolbar is the only one that can contain inline tags
+                if ($toolbarName === 'text') {
+                    continue;
+                }
+
+                $buttons = $toolbarContent['buttons'];
+                foreach ($buttons as $buttonName => $buttonConfig) {
+                    if (in_array($buttonName, $customTags, true)) {
+                        throw new InvalidConfigurationException(
+                            sprintf(
+                                'Toolbar "%s" configured in "%s" cannot contain Custom Tag "%s". Inline Custom Tags are not allowed in Toolbars other than "%s".',
+                                $toolbarName,
+                                $siteAccess,
+                                $buttonName,
+                                'text'
+                            )
+                        );
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * @param array $availableSiteAccesses
+     * @param ContainerBuilder $container
```

Comment:
CS: either FQCN or actually you can skip strictly defined param & return types if there's no extra information for that parameter (so in this instance @return can stay since it provides extra description).

Side: PhpStorm automatic code analysis not reporting issue is a good indication whether this something a correct choice. Suggested PhpStorm inspection profile for that you can find in ezsystems/ezplatform-code-style#1

Reply:
Missed this one, thanks.